### ISSUE TITLE
Improve SQL

### ISSUE THREAD

**Issue by HiveProject2021** [scotopic/silo-wallet#3]
dbcursor.execute("SELECT * FROM coin_record WHERE puzzle_hash=?", (puzzle_hash,))
Should be:
dbcursor.execute("SELECT * FROM coin_record WHERE spent=0 and puzzle_hash=?", (puzzle_hash,))

1 Just balance for address, only need to calucate the unspent records.
2 puzzle_hash not 0x begin
3 different forks have different UNITS_OF_MEASUREMENT e.g. ChiaDoge only have 9, not the 12

**Comment by scotopic** [comment]
Thanks for the feedback.

> dbcursor.execute("SELECT * FROM coin_record WHERE puzzle_hash=?", (puzzle_hash,))
> Should be:
> dbcursor.execute("SELECT * FROM coin_record WHERE spent=0 and puzzle_hash=?", (puzzle_hash,))

The reason it's `spent=0` is because  I separate the sent form unsent later in the loop -> https://github.com/scotopic/silo-wallet/blob/46fbcbd058c5998e72b16bf5402a1657d5636c8a/silo.py#L163 This is technically incomplete in case there are multiple addresses involved but works for simple cases. I'm working on beefing this up.

> 1 Just balance for address, only need to calucate the unspent records.

Yup this takes care of spent/unspent https://github.com/scotopic/silo-wallet/blob/46fbcbd058c5998e72b16bf5402a1657d5636c8a/silo.py#L163

> 2 puzzle_hash not 0x begin

Are you saying I should prepend `0x` of this output `Searching for puzzle_hash: 0xPUZZLE_HASH`? In SQL db the puzzle_hashes are not stored with `0x`.

> 3 different forks have different UNITS_OF_MEASUREMENT e.g. ChiaDoge only have 9, not the 12

* Hmm just checked GreenDoge ( previously ChiaDoge ) and DogeChia both use 12
* I see Chives uses 10 ** 8 ( now fixed in -> https://github.com/scotopic/silo-wallet/commit/8af378c43ba46313e8cbb1ff39499da2614e00c2 )
* I fixed ChiaRose already https://github.com/scotopic/silo-wallet/commit/0a6f1f029d734c5b87323fa44e7ac23cb0f68c9d and deployed in v0.1.0+

**Comment by ageorge95** [comment]
I agree with @scotopic, points 1 and 2 are not relevant for the current scripts logic (1- if implemented they would remove a current functionality, to see spend coins and 2 is wrong), and point 3 was fixed in https://github.com/scotopic/silo-wallet/commit/0a6f1f029d734c5b87323fa44e7ac23cb0f68c9d.

I think this issue can be closed.
